safe_transaction_service/history/indexers/erc20_events_indexer.py

```diff
@@ -210,7 +217,32 @@ def get_almost_updated_addresses(
 
         logger.debug("%s: Retrieving monitored addresses", self.__class__.__name__)
 
-        addresses = set(self.database_queryset.all().values_list("address", flat=True))
+        last_checked: Optional[datetime.datetime]
+        if self.addresses_cache:
+            # Only search for the new addresses
+            query = self.database_queryset.filter(
+                created__gte=self.addresses_cache.last_checked
+            )
+            addresses = self.addresses_cache.addresses
+            last_checked = self.addresses_cache.last_checked
+        else:
+            query = self.database_queryset.all()
+            addresses = set()
+            last_checked = None
+
+        for created, address in query.values_list("created", "address").order_by(
+            "created"
+        ):
+            addresses.add(address)
+
+        try:
+            last_checked = created
```

Comment:
We always try to set `last_checked` here. Would it make sense to remove the respective assignments in line 227 and 231? We can also set the initial value of `last_checked` to `None` in line 220.

Reply:
I thought about it too, but I feel it's more clear as it is right now, as it's clear that if cache is in place last_checked is the last element in cache, and otherwise is None